### PR TITLE
Create gin-instrumentation-with-metric.yml

### PR DESCRIPTION
**Pull request by Cyprinus12138**
Hi I implemented a instrumation for goinc-gin with metric.
I would like to add it to the registry of opentelemetry.io, so that more gin users would be able to benefit.

### CODE REVIEW

**Review by cartermp — APPROVED** [review]

**Comment by cartermp** [comment]
@Cyprinus12138 unfortunately this fails due to the latest docsy changes:

```
Submodule 'themes/docsy' (https://github.com/cncf/docsy.git) registered for path 'themes/docsy'
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/community'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/opamp-spec'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/opentelemetry-go'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/opentelemetry-proto'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/opentelemetry-specification'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/content-modules/semantic-conventions'...
Cloning into '/home/runner/work/opentelemetry.io/opentelemetry.io/themes/docsy'...
Submodule path 'content-modules/community': checked out 'f16a58e11ca1[49](https://github.com/open-telemetry/opentelemetry.io/actions/runs/8464934306/job/23190582362#step:9:50)82ce5c3cf375c2384422dfb3ac'
Submodule path 'content-modules/opamp-spec': checked out '0360da84c3009866b7af702606f7480eb72da6d3'
Submodule path 'content-modules/opentelemetry-go': checked out 'eabcef4c2da6149b7e6dbbb8b7294402fcc287c1'
Submodule path 'content-modules/opentelemetry-proto': checked out '4ca4f0335c63cda7ab31ea7ed70d6553aee14dce'
Submodule path 'content-modules/opentelemetry-specification': checked out '3985e212f0ef5439daf9b10f0ee490e349c69368'
Submodule path 'content-modules/semantic-conventions': checked out 'cafda7127683b7f667e27cdbd3220[51](https://github.com/open-telemetry/opentelemetry.io/actions/runs/8464934306/job/23190582362#step:9:52)0b6f998c9'
fatal: remote error: upload-pack: not our ref b077a74d38c833586361e7d735672166818a64cf
fatal: Fetched in submodule path 'themes/docsy', but it did not contain b077a74d38c833586361e7d735672166818a64cf. Direct fetching of that commit failed.
npm ERR! code 1
npm ERR! path /home/runner/work/opentelemetry.io/opentelemetry.io
npm ERR! command failed
npm ERR! command sh -c npm run seq -- get:submodule _prepare:docsy

npm ERR! A complete log of this run can be found in: /home/runner/.npm/_logs/2024-03-28T09_09_34_595Z-debug-0.log
```

In your branch, you may need to reinstall: `npm install`

**Comment by Cyprinus12138** [comment]
@cartermp hi master, thanks for your advice! I've update my code accrodingly.

File: data/registry/gin-instrumentation-with-metric.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# cSpell:ignore # add words unknown to cspell, remove this line if not required
```
Comment (cartermp) [inline data/registry/gin-instrumentation-with-metric.yml:1]:
```suggestion
# cSpell:ignore gonic Cyprinus
```

**Review by cartermp — APPROVED** [review]
This should be what fixes it!

**Review by cartermp — APPROVED** [review]
This should be what fixes it!

**Pull request merged**